### DISCUSSION TITLE
Multiple combinations with same seed?

### DISCUSSION THREAD

**Discussion by lowrykun** [adieyal/sd-dynamic-prompts#374] [Q&A]
I'm interested in running the same prompt multiple times, but with the same seed per batch. I haven't found this option.

For example: "A {cat|dog} lying on a bed". If I run this once I will get 2 images. I would like to run this prompt multiple times with a random seed set per batch.

Currently the options seem to be fixed seed or not. If I use Combinatorial batches with a random seed and 'Fixed seed' option selected, then the random seed is selected, then used for all images. Depending on the sampling method, this will typically result in the same images:
Batch 1, Seed 123: A cat lying on a bed
Batch 1, Seed 123: A dog lying on a bed
Batch 2, Seed 123: A cat lying on a bed
Batch 2, Seed 123: A dog lying on a bed

Where 123 was the randomly selected seed.

If I remove the 'Fixed seed' option, then a random seed is generated the first time, then sequential seed are use (incremented by 1).
Batch 1, Seed 123: A cat lying on a bed
Batch 1, Seed 124: A dog lying on a bed
Batch 2, Seed 125: A cat lying on a bed
Batch 2, Seed 126: A dog lying on a bed

I would like the option to generate a random seed for each batch:
Batch 1, Seed 123: A cat lying on a bed
Batch 1, Seed 123: A dog lying on a bed
Batch 2, Seed 456: A cat lying on a bed
Batch 2, Seed 456: A dog lying on a bed

I've tried the various options including "Batch Count" and "Unlink seed from prompt" but haven't been able to accomplish this. Does anyone have any thoughts?

**Comment by adieyal** [comment]
I just merged a fix for this - see [https://github.com/adieyal/sd-dynamic-prompts/pull/356](https://github.com/adieyal/sd-dynamic-prompts/pull/356) and v2.10.5
